crypto/src/main/java/org/springframework/security/crypto/key/CryptoKey.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.crypto.key;
+
+import java.security.Key;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import javax.crypto.SecretKey;
+
+/**
+ * A holder of a {@code java.security.Key} used for cryptographic operations.
+ *
+ * @param <K> the type of {@code java.security.Key}
+ * @author Joe Grandja
+ * @since 5.5
+ * @see CryptoKeySource
+ */
+public class CryptoKey<K extends Key> {
+
+	private final K key;
+
+	private final String id;
```

Comment:
This wouldn't work very well for SAML, which doesn't have a notion of selecting credentials by id. I wonder if it would be more appropriate living in `oauth2-jose` and being called `Jwk`.